src/libstd/sys/unix/fd.rs

```diff
@@ -18,6 +18,27 @@ use sys::cvt;
 use sys_common::AsInner;
 use sys_common::io::read_to_end_uninitialized;
 
+#[cfg(target_os = "android")]
+use super::android::{cvt_pread64, cvt_pwrite64};
+#[cfg(any(target_os = "linux", target_os = "emscripten"))]
+use libc::{pread64, pwrite64, off64_t, ssize_t};
+#[cfg(not(any(target_os = "linux", target_os = "emscripten", target_os = "android")))]
+use libc::{pread as pread64, pwrite as pwrite64, off_t as off64_t, ssize_t};
+
+#[cfg(not(target_os = "android"))]
+unsafe fn cvt_pread64(fd: c_int, buf: *mut c_void, count: size_t, offset: off64_t)
+    -> io::Result<ssize_t>
+{
+    cvt(pread64(fd, buf, count, offset))
+}
+
+#[cfg(not(target_os = "android"))]
+unsafe fn cvt_pwrite64(fd: c_int, buf: *const c_void, count: size_t, offset: off64_t)
+    -> io::Result<ssize_t>
+{
+    cvt(pwrite64(fd, buf, count, offset))
+}
```

Comment:
Could these functions and imports be defined locally in the `read_at` and `write_at` functions below? (the android fallbacks should stay in the android module, however.